addons/mail/static/src/models/file_uploader.js

```diff
@@ -102,6 +102,7 @@ registerModel({
                 (this.activityView && this.activityView.activityBoxView.chatter)
             ); // save before async
             const activity = this.activityView && this.activityView.activity; // save before async
+            const channel = this.threadViewTopbarOwner && this.threadViewTopbarOwner.thread.channel; // save before async
```

Comment:
Function is long and hard to maintain.
We'll need to split eventually.
In the meantime, prefer clear variable name that makes context clear.
Here it's the channel in the context of file uploading in thread topbar. Would be best to see `topbar` at least in the name.